tests/test_component.py

```diff
@@ -438,3 +439,52 @@ def test_instances_of_component_do_not_share_slots(self):
             </custom-template>
         """,
         )
+
+
+class TestComponentAsView(SimpleTestCase):
+    def test_component_view_get_request(self):
+        class MyComponent(component.Component):
+            template_name = "simple_template.html"
+
+            def get(self, request, *args, **kwargs):
+                return HttpResponse(self.render({"variable": "test"}))
+
+        comp = MyComponent()
+
+        request = HttpRequest()
+        request.method = "GET"
+        response = comp.dispatch(request)
+        self.assertEqual(response.status_code, 200)
+        self.assertIn(
+            "Variable: <strong>test</strong>", response.content.decode()
+        )
+
+    def test_component_view_post_request(self):
+        class MyComponent(component.Component):
+            template_name = "simple_template.html"
+
+            def post(self, request, *args, **kwargs):
+                return HttpResponse(self.render({"variable": "test"}))
```

Comment:
Maybe add some simple post data handling here to make the test make a little more sense?